XenModel/Actions/CrossConnectionParallelAction.cs

```diff
@@ -0,0 +1,158 @@
+﻿/* Copyright (c) Citrix Systems Inc. 
```

Comment:
I think we should consider whether we have too much

1. functionality duplication
2. code duplication

between this class and ParallelAction, and between this class and MultipleActionLauncher. Do we really need all of them, or could this class replace some others?